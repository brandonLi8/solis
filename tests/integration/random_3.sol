# Copyright Â© 2022-2023 Brandon Li. All rights reserved.

##
Integration test of a randomly generated program.
NOTE: this file was auto-generated with the `random_program_gen.py` script.
##

let a1: bool = (!((-(608)) >= ((791) / (-((-(408)) % (-(406))))))) == (true);
false;
let b1: float = 12.585806996863566;
21.873957222192512;
40;
a1;
(-((b1) / (902))) - (-(b1));
763;
-(814);
let c1: int = if !(true) {
  (531) + (-(532));
  -(if true {
    let d1: bool = !(a1);
    509;
    let e1: bool = !((294) >= (341));
    -(523)
  } else if !(false) {
    true;
    let f1: int = 223;
    let g1: float = 14.13040434952576;
    f1
  } else if a1 {
    -(14.9016385762827);
    -(40.93052529105796);
    !(false);
    let h1: int = -(304);
    let i1: float = -(b1);
    -(29.9644588677921);
    h1
  } else if !(a1) {
    let j1: bool = !((33.66838993123144) != (2.6756465606177158));
    !(((-((557) / (196))) * ((-((-(914)) / (-(if !(false) {
      !(true);
      -(37.00101613663182);
      true;
      -(263)
    } else if !(if (-(884)) >= (b1) {
      -((b1) * ((-(62)) - (77)));
      b1;
      let k1: float = b1;
      !(false)
    } else if j1 {
      35.38659465142728;
      j1
    } else if !(j1) {
      let l1: int = -(115);
      let m1: bool = !(false);
      let n1: int = -(l1);
      let o1: float = b1;
      let p1: float = 18.966920139257315;
      let q1: int = -(38);
      !((-(n1)) == (380))
    } else {
      let r1: int = (-((-(735)) - (428))) % (-(944));
      -(r1);
      let s1: int = (-((-(r1)) % (-(250)))) % (-(r1));
      -(b1);
      j1
    }) {
      a1;
      let t1: float = -(b1);
      -((973) + (-(685)))
    } else if !(false) {
      let u1: int = -(360);
      27.383525736663753;
      let v1: int = -((u1) / (-(u1)));
      let w1: int = -(504);
      -((316) % (-((12) + (-(429)))));
      let x1: float = -(b1);
      false;
      v1 + u1 + w1
    } else {
      -(8);
      let a2: bool = !(false);
      let b2: bool = !(j1);
      7.19588875142734;
      let c2: float = -(25.684529548905093);
      let d2: float = c2;
      38
    })))) % (-(242)))) < (-(288)));
    let e2: int = -(91);
    e2
  } else {
    let f2: float = -(41.028333750453214);
    let g2: float = 16.022278596222314;
    44.75515274367315;
    let h2: bool = !((true) == (false));
    -(f2);
    37.30969046488475;
    -(990)
  });
  615;
  false;
  -(44.21082444610468);
  (355) + (39)
} else if !(a1) {
  let i2: int = 580;
  -(825);
  let j2: float = b1;
  -(27.914384483070055);
  (-(198)) == (-(i2));
  -(b1);
  let k2: float = 3.195811656189911;
  let l2: float = k2;
  i2
} else if !((false) != (!(a1))) {
  -(511);
  let m2: float = 3.967314404842017;
  if a1 {
    let n2: int = -((-(187)) * (92));
    -(b1);
    (b1) == (-(b1));
    n2
  } else if a1 {
    69;
    (a1) != (!(true));
    !(false);
    -(b1);
    -(b1);
    -(b1);
    -(317)
  } else {
    47.0013646852375;
    -(930)
  };
  144;
  515
} else if !((-(447)) > (-(498))) {
  -(194)
} else if !(true) {
  !(a1);
  let o2: float = -(0.03945646117860316);
  let p2: int = (-(92)) / ((-(272)) + (-(340)));
  let q2: int = -(561);
  let r2: bool = a1;
  -((q2) + (-(o2)));
  -(((-(p2)) - (-(935))) + ((292) / (-(713))));
  (209) / (b1);
  p2
} else if true {
  -(b1);
  let s2: int = 219;
  -((-(s2)) % (s2));
  let t2: float = -(44.86859953826968);
  s2
} else {
  a1;
  1.0539073941616677;
  353;
  let u2: float = b1;
  (913) * (703);
  let v2: bool = !(a1);
  !(v2);
  !(true);
  -(260)
};
let w2: bool = false;
-(c1);
!(w2);
-(b1);
6.023050631687091;
(-(c1)) / (-(44.916263598949286));
17.97816542888929;
49.370023054492165;
let x2: float = b1;
b1;
let a3: float = -(((-(568)) / (-(6.450571443168393))) * (-(x2)));
let b3: float = (-(a3)) + (a3);
-(43.78263176614534);
let c3: float = b1;
let d3: bool = !(false);
let e3: bool = w2;
130;
-((b1) + (564));
let f3: bool = w2;
let g3: float = 4.14891340753622;
-(g3);
let h3: bool = !(false);
let i3: bool = !(false);
-(570);
c3;
-(c1);
let j3: bool = !(a1);
-(32.660924714747516);
c1;
let k3: float = 4.250015341962755;
!((!((c3) == (a3))) != (!((c1) >= (923))));
let l3: int = 292;
let m3: bool = !(true);
let n3: int = -(c1);
-(682);
let o3: bool = !((-(l3)) == ((345) / ((-(845)) + (c1))));
let p3: bool = true;
let q3: float = -(10.114765591479518);
b1;
let r3: float = -((-(807)) * (-(b1)));
let s3: float = -(31.27469470534515);
let t3: bool = !(f3);
n3;
-(46.21884144678037);
-(32.220129199477526);
true;
let u3: bool = !(a1);
let v3: int = -(n3);
-(4.042165460798574);
-(558);
let w3: bool = d3;
let x3: bool = (-(535)) < (-(409));
43.91320985104623;
-(a3);
-(b3);
!(t3);
let a4: float = -(q3);
c1;
10.655004614263635;
let b4: float = q3;
let c4: bool = x3;
true;
true;
let d4: bool = !(false);
let e4: bool = !(true);
let f4: float = b3;
let g4: bool = !(t3);
n3;
a1;
let h4: float = 34.82717830561924;
let i4: bool = !(false);
(-(84)) - (202);
-((c1) / (-(if t3 {
  let j4: bool = true;
  -((q3) / (24.64319079811106));
  true;
  -(15.77623602687688);
  let k4: int = 430;
  n3 + l3 + c1 + v3 + k4
} else if !(g4) {
  let l4: bool = !(a1);
  -(a3);
  478;
  let m4: bool = true;
  (x2) + (c1);
  -(r3);
  (-((-(759)) % (309))) >= (-(v3));
  v3 + l3
} else if p3 {
  !((-(647)) <= (-(v3)));
  76;
  let n4: int = 701;
  let o4: float = g3;
  v3;
  !(false);
  let p4: float = 35.114935171433416;
  v3 + n4 + n3
} else {
  v3 + l3 + n3
})));
let q4: bool = !(true);
!((39.74107246188393) >= ((42.887276161044795) / (x2)));
-(v3);
i3;
39.33181899443129;
!(false);
-(42.67492514580086);
let r4: float = g3;
(-(x2)) / (47.56700659992388);
846;
p3;
b1;
-(n3);
x2;
let s4: bool = o3;
!(c4);
r4;
let t4: bool = true;
let u4: int = l3;
!(false);
let v4: bool = !(t3);
let w4: int = (618) / (v3);
-(69);
!(false);
let x4: int = u4;
!(i3);
let a5: float = (q3) / (-(556));
let b5: float = -(b3);
let c5: bool = !(d3);
let d5: bool = !((-(941)) < (-(573)));
let e5: int = 127;
l3;
let f5: bool = o3;
let g5: bool = !(u3);
-(f4);
!(q4);
-(4.742773580594466);
let h5: float = 0.2262447867434886;
true;
21.619913192284077;
-(x2);
!((f4) == (-(14.458203779676067)));
let i5: float = -(29.494815937225138);
let j5: bool = true;
d5;
-(b5);
x4;
let k5: int = 787;
q3;
44.28672522145766;
let l5: int = w4;
let m5: bool = !(j5);
let n5: float = f4;
m5;
-(h5);
let o5: bool = false;
let p5: float = 5.168389824148827;
-(34.70611211124912);
-(470);
if !(false) {
  664;
  !((!(o3)) == (false));
  -(313);
  !(o5)
} else if !((-((465) * (45.00149581616694))) != (-(39.612542401260455))) {
  let q5: bool = !((351) <= (192));
  878;
  -(26.23821308073784);
  let r5: float = -(30.254203038827217);
  true
} else if true {
  633;
  let s5: float = (-(k5)) - (5.698589912032677);
  -((-((-(u4)) - (47.91014935359979))) + (-(g3)));
  1.5413724703800884;
  -(a4);
  true
} else {
  let t5: float = c3;
  false
};
!(g4);
-(l3);
-((-(l3)) * (q3));
let u5: bool = !(false);
-(38.861701328783454);
-((-(n3)) / (262));
b4 + r4 + a5 + f4 + i5 + a4 + c3 + r3 + n5 + k3 + r4