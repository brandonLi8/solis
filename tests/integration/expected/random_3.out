-4556560304651507786